### PR TITLE
 feat: add create profile to onboarding

### LINKED ISSUE #15022: Onboarding/ Fresh Install - create profile page
Designs:
https://www.figma.com/file/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=2241%3A250700&t=fD9St3RpvL2wULMb-0

Implement the initial screens and put the ui components in place.

Continue button is floating and should always be visible even when keyboard is displaying.
- If the screen is small and can not fit all content the screen should be scrolling. Align with design team to make sure of how to implement this.

Include validation for profile input.

Depends on: https://github.com/status-im/status-mobile/issues/15020
and
https://github.com/status-im/status-mobile/issues/15021

### PR DESCRIPTION
**Pull request by J-Son89**
fixes: https://github.com/status-im/status-mobile/issues/15022

Allows user to set their custom color, user name and their profile picture.

For now it is using the old profile picture code but this will be updated in future work to a new design.

Additionally on ios there is a flickering issue which happens when the image is set:
https://github.com/ivpusic/react-native-image-crop-picker/issues/1658

To fix this we need to upgrade the underlying dependency and will get done outside of this pr.

**TO TEST**
Go to the new Generate Keys flow.
You should be then able to set your profile picture, user name and custom color and continue navigation until you generate keys and then land on the home page of the app.

Should be able to create a profile with or without a profile picture

### CODE REVIEW

**Comment by du82** [comment]
This top bar should match the translucent color below it.
referenced: #15362

Screenshot_2023-03-17-08-31-14-92_65ad86f86fd2321a85272a99323da10f

**Comment by J-Son89** [comment]
> This top bar should match the translucent color below it. referenced: #15362
> 
> Screenshot_2023-03-17-08-31-14-92_65ad86f86fd2321a85272a99323da10f

Thanks @du64, it will get sorted 👍

File: src/quo2/components/inputs/title_input/view.cljs
Comment (cammellos) [inline src/quo2/components/inputs/title_input/view.cljs]: lint

**Review by cammellos — APPROVED** [review]

File: src/quo2/components/inputs/title_input/view.cljs
Comment (J-Son89) [reply src/quo2/components/inputs/title_input/view.cljs]: done

**Comment by churik** [comment]
### ISSUE 1: profile image is not set in the profile if it was set during "Generate keys" flow

**Steps:**
1) "Generate new key"
2) Set profile image, name, color
3) Finish flow and check the user profile

**Expected result:**
can see user picture

**Actual result:**
IMAGE 2023-03-21 13:34:18

**OS:** IOS, Android


### ISSUE 2: White top bar on IOS
The issue is not reproducible on nightly 21/03/23
Monosnap 2023-03-21 13-35-18

**OS:** IOS 16 (Iphone mini)
 
 
 ### Question 3:  https://github.com/status-im/status-mobile/issues/15362 will be fixed apart, right?

File: src/quo2/foundations/colors.cljs
Comment (Parveshdhull) [inline src/quo2/foundations/colors.cljs:242]: Please add line

File: src/status_im/multiaccounts/login/core.cljs
Comment (Parveshdhull) [inline src/status_im/multiaccounts/login/core.cljs:484]: @smohamedjavid this pr addresses this flow, so can also be used for sign in too.

File: src/status_im2/contexts/onboarding/create_password/view.cljs
Comment (Parveshdhull) [inline src/status_im2/contexts/onboarding/create_password/view.cljs:38]: leftover?

File: src/status_im2/contexts/onboarding/create_profile/view.cljs
Comment (Parveshdhull) [inline src/status_im2/contexts/onboarding/create_profile/view.cljs]: Should we move them to constants file?

File: src/status_im2/contexts/onboarding/create_profile/view.cljs
Comment (Parveshdhull) [inline src/status_im2/contexts/onboarding/create_profile/view.cljs]: At most places we have :default color as :primary, but it looks like figma colors file don't even have primary color anymore 
We probably need to change this too on all the places. (probably in separate PR)

File: src/status_im2/contexts/onboarding/generating_keys/view.cljs
Comment (Parveshdhull) [inline src/status_im2/contexts/onboarding/generating_keys/view.cljs]: move to translations file

**Review by Parveshdhull — APPROVED** [review]
some minor suggestions, apart from them lgtm

File: src/status_im2/contexts/onboarding/create_profile/view.cljs
Comment (J-Son89) [reply src/status_im2/contexts/onboarding/create_profile/view.cljs]: probably this should just be in constants or something like that.

File: src/status_im2/subs/root.cljs
Comment (briansztamfater) [inline src/status_im2/subs/root.cljs:273]: Could we use `onboarding/profile` instead of `onboarding-2`? I feel that we will need / want to change `onboarding-2/...` in the future and maybe we can do it now

Comment (J-Son89) [reply src/status_im2/subs/root.cljs:273]: this is intentional to have a clear separation of the old onboarding and new onboarding subs. The idea being that when we move away from the old onboarding that we can completely nuke all `onboarding/` subs etc and rename all new `onboarding-2` subs to `onboarding`. 
Sure we could do it now by changing the old to something like `onboarding-old/` instead but this would be a lot of extra changes, noise and work and potentially brings in bugs from missing some parts.
@cammellos and I agreed upon this. 
Can be changed of course but imo it's the better option as is 👍

File: src/status_im2/contexts/onboarding/create_password/view.cljs
Comment (J-Son89) [reply src/status_im2/contexts/onboarding/create_password/view.cljs:38]: no this is just placeholder until @ulisesmac pr for password page is merged

File: src/status_im2/subs/root.cljs
Comment (briansztamfater) [reply src/status_im2/subs/root.cljs:273]: Understood. I did a quick and vague search I haven't seen other `onboarding/` subs or db keys so that's why I don't see necessary to use the `-2` suffix, but it is okay for me anyway

**Comment by J-Son89** [comment]
> ### ISSUE 1: profile image is not set in the profile if it was set during "Generate keys" flow
> **Steps:**
> 
> 1. "Generate new key"
> 2. Set profile image, name, color
> 3. Finish flow and check the user profile
> 
> **Expected result:** can see user picture
> 
> **Actual result:** IMAGE 2023-03-21 13:34:18
> 
> **OS:** IOS, Android
> 
> ### ISSUE 2: White top bar on IOS
> The issue is not reproducible on nightly 21/03/23 Monosnap 2023-03-21 13-35-18
> 
> **OS:** IOS 16 (Iphone mini)
> 
> ### Question 3: #15362 will be fixed apart, right?

@churik I have fixed Issue 1 on ios and android 👍 Issue 2 is also fixed here and Question 3 will be handled separately. 👍

**Comment by churik** [comment]
### ISSUE 4: cannot read property 'toUpperCase' in case of starting if the name is starting from any uppercase with space at the beginning

**Steps:**
1) "Generate new key"
2) on the name filed type space and uppercase, i.e. `    F`
**Expected result:**
can proceed
**Actual result:**


https://user-images.githubusercontent.com/4557972/226680347-79762fd0-1f11-4615-9801-df77482eaed5.mp4


IMAGE 2023-03-21 17:41:04


**OS:** IOS, Android

**Comment by J-Son89** [comment]
> ### ISSUE 4: cannot read property 'toUpperCase' in case of starting if the name is starting from any uppercase with space at the beginning
> **Steps:**
> 
> 1. "Generate new key"
> 2. on the name filed type space and uppercase, i.e. `    F`
>    **Expected result:**
>    can proceed
>    **Actual result:**
> 
> IMAGE 2023-03-21 17:41:04
> 
> **OS:** IOS, Android

@churik, I fixed this whitespace issue and added some basic validation to the profile name input as per the designs too 👍

**Comment by churik** [comment]
@J-Son89 

### ISSUE 5 Font is too white and invisible in black/white color


**OS:** IOS, Android
IMAGE 2023-03-22 12:43:00

IMAGE 2023-03-22 12:42:56

### ISSUE 6 Difference between design and actual implementation 
photo_2023-03-22 12 48 42 jpeg 2023-03-22 12-49-18

- 6.1 The placeholder font color is not aligned
- 6.2 The cursor color is not aligned
- 6.3 No opacity either on IOS or on Android
- 6.4 'Continue' button is overlapped by the keyboard and the view is not scrollable 

### ISSUE 7 Button continue is still available when validation is not passed
Therefore, can create multi-account with invalid names.

**Steps:**
- set name with emoji 
- finish the flow

Besides, there is no specification of what is considered 'generic' names. Per design  from https://www.figma.com/file/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=2241%3A250700&t=fD9St3RpvL2wULMb-0 it is only `Ethereum`, but I guess it is not enough. But it can be addressed separately.

Let me know if smth from mentioned is already discussed and if I need to ping designers about acceptable differences for now.

Thank you!

**Comment by J-Son89** [comment]
@churik, is it possible I follow up addressing those issues? Getting this pr in will unblock some of the developers on the other pages.

To answer each item:
ISSUE 5 Font is too white and invisible in black/white color
- Design team have not given us a clear explanation of how this is to be handled for black and white yet so we are unable to build it correctly. IMO it is probably better that we remove this option, will this discuss with design team but for now it will not be working correctly as long as it is selectable.

ISSUE 6 & 7 should be addressed immediately but I can do it in a follow up issue/pr.

Validation is only general at this point as we do not have an extensive list of common names etc that we need to remove. 

cc: @cammellos

**Comment by churik** [comment]
I'll open follow-up, no worries! thank you for your work, ready to be merged

**Pull request merged**
